### DISCUSSION TITLE
Vesting Contract

### DISCUSSION THREAD

**Discussion by vigh18** [Agoric/agoric-sdk#5683] [Q&A] [answered]
Let assume that Agoric's issuer, mint, brand trio is the way how can anyone create its own tokens which is analogy to ERC20. I can see a lot of use cases where the token owner want to distribute his tokens to other people according some specified vesting schedule by some trustless way. And the vesting contracts can exactly do that. We can take an open zeppelin as a reference. https://github.com/OpenZeppelin/token-vesting-ui

**Comment by dckc** [comment]
good question... we started discussing it...

**Comment by trentlapinski** [comment]
Definitely something we’re looking into.

I see that project is no longer maintained, was there anything specific you liked about it? How could we make it better?

**Comment by dckc** [comment]
@JimLarson that ERTP-based design you told us about for things like vesting... are there any materials to share here? Perhaps from when you presented it to the cosmos-sdk crowd?

**Reply by JimLarson** [reply, marked as answer]
Here's the notes I had on ERTP-based staking and vesting: https://drive.google.com/file/d/1FmJGhnnf5GOLZ3YF3StjEHSDMSwU63Vg/view?usp=sharing

I also translated the same dynamics into the Cosmos-SDK world, and you could create a single interoperable system that spanned ERTP and Cosmos: https://docs.google.com/presentation/d/1wY0hZutddeM7Lmn1leboGAQj4qe0iLVqUv4a7JtQ1j4/edit?usp=sharing

**Reply by dckc** [reply]
Thanks, @JimLarson !
